encoding.go

```diff
@@ -0,0 +1,39 @@
+package sprig
+
+import (
+	"encoding/base32"
+	"encoding/base64"
+	"net/url"
+)
+
+func base64encode(v string) string {
+	return base64.StdEncoding.EncodeToString([]byte(v))
+}
+
+func base64decode(v string) string {
+	data, err := base64.StdEncoding.DecodeString(v)
+	if err != nil {
+		return err.Error()
+	}
+	return string(data)
+}
+
+func base32encode(v string) string {
+	return base32.StdEncoding.EncodeToString([]byte(v))
+}
+
+func base32decode(v string) string {
+	data, err := base32.StdEncoding.DecodeString(v)
+	if err != nil {
+		return err.Error()
+	}
+	return string(data)
+}
+
+func urlEncode(v string) string {
+	return url.QueryEscape(v)
```

Comment:
Can you explain the use case for `urlEncode`?

To provide a little context...
1. `url.QueryEscape` isn't meant for a full URL. Instead, it's meant for the Query string part of it.
2. The `html/template` package does automatic URL escaping. If you want to learn more about that see the [`escape.go`](https://golang.org/src/html/template/escape.go) and [`url.go`](https://golang.org/src/html/template/url.go) files.

So, this function (once for full URLs) would need a case outside of html templates.

Reply:
> url.QueryEscape isn't meant for a full URL. Instead, it's meant for the Query string part of it.

This is both true and false. I think what you're saying is that if you encode `http://www.example.com`, you're going to end up with `http:%2F%2Fwww.example.com`, and you're absolutely right-- but it's probably not correct that nobody ever wants to do that. Imagine, for instance, the case where a URL needs to be passed as a query parameter within another url.

Although the above isn't my actual use case. It's just a good example. Mine real case is that I want to URL encode a password that some off-the-shelf software concatenates into a connection string without (shame on them) properly URL encoding it themselves.

I think whether we are talking about sprig template functions or just plain `url.QueryEscape` in go, _what_ you do with it is relative to one's use case. It's probably not for us to decide how it's used, but rather just to provide the capability.

Reply:
I would need this to be able to create custom urls.

My goal is to create URL which would redirect to prometheus, e.g.:

```
https://prometheus.company.com/graph?g0.range_input=1h&g0.expr={{ urlEnc .query }}&g0.tab=1
```